src/mongoc/mongoc-client.c

```diff
@@ -1730,6 +1731,16 @@ _mongoc_client_command_with_opts (mongoc_client_t *client,
    parts.is_read_command = (mode & MONGOC_CMD_READ);
    parts.is_write_command = (mode & MONGOC_CMD_WRITE);
 
+   command_name = _mongoc_get_command_name (command);
+
+   if (!command_name) {
+      bson_set_error (error,
+                      MONGOC_ERROR_COMMAND,
+                      MONGOC_ERROR_COMMAND_INVALID_ARG,
+                      "Empty command document");
+      GOTO (err);
```

Comment:
Note: I deferred this check until after `mongoc_cmd_parts_init()`, as the end of this function always calls `mongoc_cmd_parts_cleanup()`. While the `parts` struct is always stack-allocated, I'm not sure how the `bson_destroy()` calls within the cleanup function would deal with empty pointers (assuming `parts` is zero-allocated).